src/components/views/elements/AppPermission.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import url from 'url';
+
+export default class AppPermission extends React.Component {
+    constructor(props) {
+        super(props);
+
+        const curlBase = this.getCurlBase();
+        this.state = {
+            curlBase: curlBase,
```

Comment:
Hmm... I really think that it is clearer like this, so you might be seeing this again (is there a style guideline for this?), but I've inlined it as suggested ;)